_posts/2022-01-23-feature-deep-dive-opensearch-sql-basic-queries.markdown

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+title:  "Feature Deep Dive: OpenSearch SQL - Basic queries"
+authors: 
+  - penghuo
+  - jadhanir
+date: 2022-01-23
+categories: 
+  - feature
+
+---
+
+In this blog, we’ll explore some of the features and capabilities supported by OpenSearch SQL. We also briefly introduce how SQL engine works internally.
+
+## What is OpenSearch SQL?
+
+OpenSearch SQL is similar to the SELECT statement in ANSI SQL, but it is purely designed for analyzing data in OpenSearch. In this document, we refer to it as OSQL (short for OpenSearch Structured Query Language). In order to make it easier for traditional SQL customers to get started, we make the OSQL syntax conform to the subset of the ANSI SQL specification. In addition, we also disclosed the unique features of OpenSearch, for example, we support [full-text search](https://github.com/opensearch-project/sql/blob/main/docs/user/beyond/fulltext.rst).
+
+## How to use OpenSearch SQL?
+
+For OpenSearch user, you could use the familiar REST interface `_plugin/sql` with your query in query body. 
+For DBMS user, you could use [JDBC](https://github.com/opensearch-project/sql/tree/main/sql-jdbc)/[ODBC](https://github.com/opensearch-project/sql/tree/main/sql-odbc) driver to connect OpenSearch domain. You could download JDBC/ODBC driver. - add download link
+For Dashboards user, you could use [OpenSearch Dashboard Query Workbench](https://opensearch.org/docs/latest/search-plugins/sql/workbench/) to easily run on-demand SQL queries and download results
+An additional OpenSearch SQL CLI tool is provided for interactive SQL execution.
+
+## Some Examples
+
+### Basic Queries
+
+*“Find error logs where response code is 404 or 503”.* 
+You could run ODQL in Query Workbench. The result is in tabular format. You could download result in JSON, JDBC, CSV and Text format. Find more examples in [here](https://github.com/opensearch-project/sql/blob/main/docs/user/interfaces/protocol.rst).
+![basic queries](/assets/media/blog-images/2022-01-30-opensearch-sql-basic-queries/basic_queries.gif){: .img-fluid}
+### Functions
+
+*“Find error logs where response code is 404 or 503, How many distinct host”* 
+You could run ODQL in Dashboards notebook also. There are 4 distinct hosts for about query,  result is in table format. More [aggregations](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/aggregations.rst) and [functions](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst) are supported.
+![functions](/assets/media/blog-images/2022-01-30-opensearch-sql-basic-queries/functions.gif){: .img-fluid}
+### Explain
+
+*“How does ODQL execute in OpenSearch”*. Let’s explain an example query. OpenSearch SQL explain endpoint returns the query execution plan. For example, the above query two operators.
+
+* ProjectOperator, execute in coordinate node, read output from child operator and project originCountry and count fields.
```

Comment:
- "ProjectOperator, execute in coordinate node, read" ...  "originCountry and count" => "\`ProjectOperator\' , executes in the coordinating node reads" ... "\`originCountry\` and \`count\`"
- should it be "child operator" or "child operation"?

Reply:
Done.
child operator is more accurate.